tests/datasets/storages/processors/test_replaced_groups.py

```diff
@@ -0,0 +1,133 @@
+from typing import Sequence
+
+import pytest
+
+from snuba import state
+from snuba.clickhouse.columns import ColumnSet
+from snuba.clickhouse.query import Query as ClickhouseQuery
+from snuba.datasets.errors_replacer import (
+    ReplacerState,
+    set_project_exclude_groups,
+    set_project_needs_final,
+)
+from snuba.datasets.schemas.tables import TableSource
+from snuba.datasets.storages.processors.replaced_groups import (
+    CONSISTENCY_ENFORCER_PROCESSOR_ENABLED,
+    PostReplacementConsistencyEnforcer,
+)
+from snuba.query.conditions import BooleanFunctions
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.logical import Query as LogicalQuery
+from snuba.redis import redis_client
+from snuba.request.request_settings import HTTPRequestSettings
+
+
+def build_in(project_column: str, projects: Sequence[int]) -> Expression:
+    return FunctionCall(
+        None,
+        "in",
+        (
+            Column(None, project_column, None),
+            FunctionCall(None, "tuple", tuple([Literal(None, p) for p in projects])),
+        ),
+    )
+
+
+@pytest.fixture
+def query() -> ClickhouseQuery:
+    return ClickhouseQuery(
+        LogicalQuery(
+            {"conditions": [("project_id", "IN", [2])]},
+            TableSource("my_table", ColumnSet([])),
+            condition=build_in("project_id", [2]),
+        )
+    )
+
+
+def setup_function() -> None:
+    state.set_config(CONSISTENCY_ENFORCER_PROCESSOR_ENABLED, 1)
+
+
+def teardown_function() -> None:
+    redis_client.flushdb()
+
+
+def test_with_turbo(query: ClickhouseQuery) -> None:
```

Comment:
These are the same test cases taken from the project extension test. Just cleaned it up a bit.